src/action/base/create_directory.rs

```diff
@@ -236,6 +289,48 @@ impl Action for CreateDirectory {
     }
 }
 
+// There are cleaner ways of doing this (eg `systemctl status $PATH`) however we need a widely supported way.
+async fn path_is_mountpoint(path: &Path) -> Result<bool, ActionErrorKind> {
+    let path_str = match path.to_str() {
+        Some(path_str) => path_str,
+        None => return Ok(false),
+    };
+
+    let output = execute_command(Command::new("mount")
+        .arg("-f") // `-f` means `--fake` not `--force`
```

Comment:
This isn't the same on darwin (`-f` is not `--fake` on darwin -- `Forces the revocation of write access when trying to downgrade a filesystem mount status from read-write to read-only.`). I think we'll want `-d` instead: `Causes everything to be done except for the actual system call.`

Reply:
Sadly `--fake` on Linux doesn't seem to work for all linuxes! Agree with the rest, great catch!